test/e2e_kubeadm/networking_test.go

```diff
@@ -0,0 +1,151 @@
+					dualStack = true
+				}
+			}
+			if _, ok := d["serviceSubnet"]; ok {
+				// If serviceSubnet is not specified, serviceSubnet cases will be skipped.
+				serviceSubnetInKubeadmConfig = true
+			}
+		}
+	})
+
+	Context("single-stack", func() {
+		Context("podSubnet", func() {
+			It("should be properly configured if specified in kubeadm-config", func() {
+				if dualStack {
+					framework.Skipf("Skipping because cluster is dual-stack")
+				}
+				if !podSubnetInKubeadmConfig {
+					framework.Skipf("Skipping because podSubnet was not specified in kubeadm-config")
+				}
+				cc := getClusterConfiguration(f.ClientSet)
+				if _, ok := cc["networking"]; ok {
+					d := cc["networking"].(map[interface{}]interface{})
+					if ps, ok := d["podSubnet"]; ok {
+						// Check that the pod CIDR allocated to the node(s) is within the kubeadm-config podCIDR.
+						nodes, err := f.ClientSet.CoreV1().Nodes().List(metav1.ListOptions{})
+						framework.ExpectNoError(err)
+						for _, node := range nodes.Items {
+							if !subnetWithinSubnet(ps.(string), node.Spec.PodCIDR) {
+								e2elog.Failf("failed due to node(%v) IP %v not inside configured pod subnet: %s", node.Name, node.Spec.PodCIDR, ps)
+							}
+						}
+					}
+				}
+			})
+		})
+		Context("serviceSubnet", func() {
+			It("should be properly configured if specified in kubeadm-config", func() {
+				if dualStack {
+					framework.Skipf("Skipping because cluster is dual-stack")
+				}
+				if !serviceSubnetInKubeadmConfig {
+					framework.Skipf("Skipping because serviceSubnet was not specified in kubeadm-config")
+				}
+				cc := getClusterConfiguration(f.ClientSet)
+				if _, ok := cc["networking"]; ok {
+					d := cc["networking"].(map[interface{}]interface{})
+					if ss, ok := d["serviceSubnet"]; ok {
+						// Get the kubernetes service in the default namespace.
+						// Check that service CIDR allocated is within the serviceSubnet range.
+						svc, err := f.ClientSet.CoreV1().Services("default").Get("kubernetes", metav1.GetOptions{})
+						framework.ExpectNoError(err)
+						if !ipWithinSubnet(ss.(string), svc.Spec.ClusterIP) {
+							e2elog.Failf("failed due to service(%v) cluster-IP %v not inside configured service subnet: %s", svc.Name, svc.Spec.ClusterIP, ss)
+						}
+					}
+				}
+			})
+		})
+	})
+	Context("dual-stack", func() {
+		Context("podSubnet", func() {
+			It("should be properly configured if specified in kubeadm-config", func() {
+				if !dualStack {
+					framework.Skipf("Skipping because cluster is not dual-stack")
+				}
+				if !podSubnetInKubeadmConfig {
+					framework.Skipf("Skipping because podSubnet was not specified in kubeadm-config")
+				}
+				// TODO(awander): Complete this as dual-stack support lands in master
+			})
+		})
+	})
+
+})
+
+// ipWithinSubnet returns true if an IP (targetIP) falls within the reference subnet (refIPNet)
+func ipWithinSubnet(refIPNet, targetIP string) bool {
+	_, rNet, _ := net.ParseCIDR(refIPNet)
```

Comment:
what happens when we cannot parse the cidr

Reply:
These funcs are only used here. The two parameters passed in are both checked in other places and are pulled from a deployed cluster. If they are somehow not in correct CIDR format, then there is likely something terribly wrong that goes beyond just these tests :)